bin/shpec

```diff
@@ -36,11 +36,30 @@ end_describe() {
 # any identifier as a function name.
 
 stub_command() {
+  local func_decl="$(get_function_declaration "$1")"
+  [ -n "${func_decl}" ] && eval "_shpec____${func_decl}"
+
   _shpec_stub_body="${2:-:}"
   eval "$1() { ${_shpec_stub_body}; }"
 }
 
-unstub_command() { unset -f "$1"; }
+unstub_command() { 
+  unset -f "$1";
+  local func_decl="$(get_function_declaration _shpec____"$1")"
+  [ -n "${func_decl}" ] && {
+    local reverted="$(echo "${func_decl}" | sed 's/_shpec____//')"
+    eval "${reverted}"
+  }
+}
+
+get_function_declaration()
+{
+  if [ "${SHELL}" = "dash" ]; then
+    type "$1" | tail -n +2
```

Comment:
Given the discussion in #89, [dash's own docs](https://wiki.ubuntu.com/DashAsBinSh#type), and our previous conversation, I'm fine keeping use of the (non-POSIX) `type` and `typeset` commands. If this becomes a problem, we can revisit (or move this feature to a plugin).